libs/moloch-v3-context/src/hooks/useProposals.tsx

```diff
@@ -1,55 +1,110 @@
+export const useProposals = (): MolochV3DaoContextProposalsType => {
+  const { daoData, daoid, daochain, graphApiKeys, setDaoData } =
+    useContext(MolochV3DaoContext);
+
+  const filterProposals = async (filter: Proposal_Filter) => {
+    if (daoid && daochain) {
+      const res = await fetchProposalsList({
+        filter: { dao: daoid, ...filter },
+        ordering: daoData?.proposals?.ordering,
+        daochain: daochain as ValidNetwork,
+        graphApiKeys,
+      });
+
+      setDaoData((prevState) => {
+        return { ...prevState, proposals: res };
+      });
+    }
   };
 
-  const filterProposals = async (filterQuery?: Proposal_Filter) => {
-    setProposals(undefined);
-    setProposalsFilter(filterQuery);
-    setProposalsPaging(defaultDaoData.proposalsPaging);
-    if (proposalsFilter === filterQuery) refreshProposals();
+  const sortProposals = async (ordering: Ordering<Proposal_OrderBy>) => {
+    if (daoid && daochain) {
+      const res = await fetchProposalsList({
+        filter: daoData?.proposals?.filter || { dao: daoid },
+        ordering: ordering,
+        daochain: daochain as ValidNetwork,
+        graphApiKeys,
+      });
+
+      setDaoData((prevState) => {
+        return { ...prevState, proposals: res };
+      });
+    }
+  };
+
+  const loadNextPage = async () => {
+    if (daoid && daochain) {
+      const res = await fetchProposalsList({
+        filter: daoData?.proposals?.filter || { dao: daoid },
+        ordering: daoData?.proposals?.ordering,
+        paging: daoData?.proposals?.nextPaging,
+        daochain: daochain as ValidNetwork,
+        graphApiKeys,
+      });
+
+      setDaoData((prevState) => {
+        const prevItems = prevState.proposals
+          ? [...prevState.proposals.items]
+          : [];
+
+        return {
+          ...prevState,
+          proposals: {
+            ...res,
+            items: res ? [...prevItems, ...res.items] : [],
+          },
+        };
+      });
+    }
+  };
+
+  const refreshProposals = async () => {
+    if (daoid && daochain) {
+      const res = await fetchProposalsList({
+        filter: { dao: daoid },
+        daochain: daochain as ValidNetwork,
+        graphApiKeys,
+      });
+
+      setDaoData((prevState) => {
+        return { ...prevState, proposals: res };
+      });
+    }
   };
 
   return {
-    proposals,
-    setProposals,
-    isProposalsLoading,
-    refreshProposals,
-    proposalsFilter,
-    setProposalsFilter,
-    proposalsSort,
-    setProposalsSort,
-    proposalsPaging,
-    setProposalsPaging,
-    proposalsNextPaging,
-    getNextPage,
-    loadMoreProposals,
+    proposals: daoData?.proposals?.items,
+    filter: daoData?.proposals?.filter,
     filterProposals,
+    sort: daoData?.proposals?.ordering,
+    sortProposals,
+    paging: {
+      current: daoData?.proposals?.previousPaging,
+      next: daoData?.proposals?.nextPaging,
+    },
+    loadNextPage,
+    refreshProposals,
```

Comment:
I think separating the concerns here was a good idea. This is the busiest one, but still way easier to read.